src/scaffolded-question/components/runtime.tsx

```diff
@@ -105,7 +105,7 @@ export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, set
         setInteractiveState={readOnly ? undefined : handleNewInteractiveState.bind(null, currentInteractive.id)}
         report={readOnly}
         logRequestData={logRequestData}
-        onUnloadCallback={readOnly ? undefined : handleNewInteractiveState.bind(null, currentInteractive.id)}
+        onUnloadCallback={report ? undefined : handleNewInteractiveState.bind(null, currentInteractive.id)}
```

Comment:
I think it is probably done this way so we don't accidentally set the interactive state when in report mode, since that is supposed to be display only.